code/game/objects/effects/effect_system/effects_foam.dm

```diff
@@ -7,159 +7,237 @@
 	icon_state = "foam"
 	opacity = FALSE
 	density = FALSE
+	gender = PLURAL
 	layer = OBJ_LAYER + 0.9
-	animate_movement = 0
-	var/amount = 3
-	var/expand = 1
-	var/metal = 0
-
-/obj/effect/particle_effect/foam/New(loc, ismetal = FALSE)
-	..(loc)
-	icon_state = "[ismetal ? "m":""]foam"
-	if(!ismetal && reagents)
-		color = mix_color_from_reagents(reagents.reagent_list)
-	metal = ismetal
+	animate_movement = NO_STEPS
+	/// How many times this one bit of foam can spread around itself
+	var/spread_amount = 3
+	/// How long it takes this to initially start spreading after being dispersed
+	var/spread_time = 0.9 SECONDS
+	/// How long it takes this, once it's spread, to stop spreading and disperse its chems
+	var/solidify_time = 12 SECONDS
+	/// Whether it reacts on or after dispersion (or both)
+	var/react_mode = FOAM_REACT_ON_DISSIPATE | FOAM_REACT_BEFORE_SPREAD
+	/// Maximum amount of reagents gained by spreading onto a foamed tile
+	var/max_amount_on_spread = 27
+	/// We will never fill a mob with more than this many units of a given reagent
+	var/max_reagent_filling = 25
+	/// Whether or not to spread at a range when spreading
+	var/spread_at_range = TRUE
+
+/obj/effect/particle_effect/foam/Initialize(mapload)
+	. = ..()
+	create_reagents(25)
```

Comment:
Should the capacity be more if we can gain more reagents than the capacity?

Reply:
Probably, but that might constitute a balance change, since it'd cause foam to carry more reagents per.

Reply:
Sounds good!